### ISSUE TITLE
WriteRequest offers constructor with mode and resourceId

### ISSUE THREAD

**Issue by boaks** [eclipse/leshan#114]
May be, I'm totaly wrong, but why does 

`org.eclipse.leshan.core.request.WriteRequest `offer a constructor with mode and resourceid (e.g. 
```
public WriteRequest(final Mode mode, final int objectId, final int objectInstanceId, final int resourceId,
            String value)
```
In my understanding of the LWM2M TS, writing to resources always use PUT  (REPLACE). So the API may lead to wrong messages, if POST (UPDATE) is used.

**Comment by sbernard31** [comment]
You're right. We need to change this since the update on resource was removed.

**Comment by sbernard31** [comment]
Fixed in master (commit 6ec8aeb570f63349674512560317143e192fada6)
